src/components/LoginPage.js

```diff
@@ -1,43 +1,124 @@
 
-function LoginPage({history, location}) {
+export class LoginPage extends Component {
+  constructor(props) {
+    super(props);
+  
+    this.state = {
+      email: null,
+      password: null
+    };
+  } 
 
-  useEffect(()=>{
-    // Redirect to home if logged in already
-    if(localStorage.getItem("logged_in")){
-      history.push('/');
+  componentDidMount() {
+    this.setState((prevState) => ({...prevState}));
+    this.checkLoggedIn();
+    const { location, socialAuth } = this.props;
+    if(location !== undefined) {
+    const base64encoded = location.search.split('&')[0].split('?code=')[1];
+    if (base64encoded) {
+      const decoded = JSON.parse(atob(base64encoded));
+      socialAuth(decoded);
+    }
+  }
+  }
+
+  componentDidUpdate() {
+    const { user, history } = this.props;
+    if (user.isLoggedIn === true ) {
+      const token = localStorage.getItem('bareFootToken');
+      if(token) {
+        history.push('/');
+      }
     }
-  },[history]);
-
-  const handleClick = () => {
-    localStorage.setItem("logged_in", true);
-    if(!location.state){
-      history.push("/");
-    }else{
-      // Redirect to page that was trying to login
-      const redirectTo = location.state.redirectTo.path;
-      history.push(redirectTo);
+  }
+
+  checkLoggedIn = () => {
+    const { history } = this.props;
+    const token = localStorage.getItem('bareFootToken');
+    if(token !== null) {
+      history.push('/');
     }
+  }
+
+  handleInput = async (e) => {
+    this.setState({
+      [e.target.name]: e.target.value
+    });
   };
+  
+  handleSubmit = (e) => {
+    const { localAuth } = this.props;
+    const { email, password } = this.state;
+    e.preventDefault();
+    const user = {
+      userEmail: email,
+      userPassword: password,
+    };
+    localAuth(user);
+  }
 
-  return (
-      <div>
-      <h6>Click Button to Login</h6>
-      <button type="button" onClick={handleClick} className="btn-sample">
-        Press To Login
-      </button>
+  render() {
+    const { password, email } =this.state;
+    return (
+      <div className="login-container">
+            <div className="local">
+                <img src={logo} alt="logo" />
+            <form className="loginForm" onSubmit={this.handleSubmit}>
+              <Input 
+                name="email"
+                inputType="email"
+                placeholder="Email"
+                onChange={this.handleInput}
+                value={email}
+                required
+              />
+              <Input
+                name="password"
+                inputType="password"
+                placeholder="Password"
+                onChange={this.handleInput}
+                value={password}
+                required
+              />
+            <div className="forgot">
+                <a  href="/">Forgot your password?</a>
+            </div>
+            <button className="btnn btnn-primary log" type="submit">Login</button>
```

Comment:
What is the reason behind using `btnn` for buttons

Reply:
Ineeded to add some more styles for the anchor tags was changing to buttons and didn't want to collide with the general buttons but it's specific to only those